### PR TITLE
Disconnect GKE operators from deprecated hooks

### PR DESCRIPTION
**Pull request by moiseenkov**
This PR refactors GKE operators through replacing deprecated hooks with refactored GKE hooks.

### CODE REVIEW

File: airflow/providers/google/cloud/operators/kubernetes_engine.py [hunk 1]
```diff
@@ -534,27 +530,28 @@ def cluster_hook(self) -> GKEHook:
         )
 
     @cached_property
-    def deployment_hook(self) -> GKEDeploymentHook:
+    def deployment_hook(self) -> GKEKubernetesHook:
         if self._cluster_url is None or self._ssl_ca_cert is None:
             raise AttributeError(
                 "Cluster url and ssl_ca_cert should be defined before using self.hook method. "
```
Comment (e-galan) [inline airflow/providers/google/cloud/operators/kubernetes_engine.py:536]: Although it is not part of your PR, I find this line a bit misleading, how about: 
```suggestion
                "Cluster url and ssl_ca_cert should be defined before using self.deployment_hook method. "
```
?

File: airflow/providers/google/cloud/operators/kubernetes_engine.py [hunk 2]
```diff
@@ -534,27 +530,28 @@ def cluster_hook(self) -> GKEHook:
...
             cluster_url=self._cluster_url,
             ssl_ca_cert=self._ssl_ca_cert,
         )
 
     @cached_property
-    def pod_hook(self) -> GKEPodHook:
+    def pod_hook(self) -> GKEKubernetesHook:
         if self._cluster_url is None or self._ssl_ca_cert is None:
             raise AttributeError(
                 "Cluster url and ssl_ca_cert should be defined before using self.hook method. "
```
Comment (e-galan) [inline airflow/providers/google/cloud/operators/kubernetes_engine.py:550]:
```suggestion
                "Cluster url and ssl_ca_cert should be defined before using self.pod_hook method. "
```
?

**Comment by MaksYermak** [comment]
LGTM and I also agree with Eugene about changing the log's message.

File: airflow/providers/google/cloud/operators/kubernetes_engine.py (hunk 1 above)
Comment (moiseenkov) [reply airflow/providers/google/cloud/operators/kubernetes_engine.py]: done

File: airflow/providers/google/cloud/operators/kubernetes_engine.py (hunk 2 above)
Comment (moiseenkov) [reply airflow/providers/google/cloud/operators/kubernetes_engine.py]: done

**Review by e-galan — APPROVED** [review]
Looks good

**Pull request closed without merging**